actionview/lib/action_view/helpers/tag_helper.rb

```diff
@@ -26,7 +26,42 @@ module TagHelper
       PRE_CONTENT_STRINGS[:textarea]  = "\n"
       PRE_CONTENT_STRINGS["textarea"] = "\n"
 
+      class TagBuilder #:nodoc:
+        include TagHelper
 
+        VOID_ELEMENTS = %w(base br col embed hr img input keygen link meta
+        param source track wbr).to_set
+
+        VOID_ELEMENTS.merge(VOID_ELEMENTS.map(&:to_sym))
+
+        def initialize(view_context)
+          @view_context = view_context
+        end
+
+        private
+          def tag_string(name, content_or_options = nil, options_or_escape = nil, escape = true, &block)
```

Comment:
Let's make use of Ruby's keyword arguments here:

```ruby
def tag_string(name, content = nil, escape_attributes: true, **options, &block)
  case
  when block_given?
    content_tag_string(name, @view_context.capture(self, &block), content, options)
  when content.is_a?(String)
    content_tag_string(name, content, options, escape_attributes)
  when VOID_ELEMENTS.include?(name)
    "<#{name}#{tag_options(content, escape_attributes)}>".html_safe
  else
    content_tag_string(name, "", options, escape_attributes)
  end
end
```

This lets us write this nicer code:

```ruby
tag.img src: 'open &amp; shut.png', escape_attributes: false
```